api/app/signals/apps/email_integrations/signal_receivers.py

```diff
@@ -3,9 +3,50 @@
 from django.dispatch import receiver
 
 from signals.apps.email_integrations import tasks
-from signals.apps.signals.managers import create_initial, update_status
+from signals.apps.email_integrations.services import MailService
+from signals.apps.signals.managers import (
+    create_initial,
+    update_signal_departments,
+    update_status,
+    update_user_assignment
+)
 
 
 @receiver((create_initial, update_status), dispatch_uid='core_email_integrations_update_status')
 def update_status_handler(sender, signal_obj, *args, **kwargs):
     tasks.send_mail_reporter.delay(pk=signal_obj.pk)
+
+
+@receiver(update_signal_departments, dispatch_uid='core_email_integrations_update_signal_departments')
+def update_signal_departments_handler(sender, signal_obj, signal_departments, prev_signal_departments, *args, **kwargs):
+    if signal_departments and prev_signal_departments:
+        if set(signal_departments.departments.all()) == set(prev_signal_departments.departments.all()):
+            return  # do not trigger when the departments field is unchanged
+
+    # TODO: make async
+    for department in signal_departments.departments.all():
+        select_related = department.user_profiles.select_related('user')
+        for profile in select_related.filter(notification_on_department_assignment=True, user__is_active=True):
```

Comment:
Naming nit: why not `profiles` or `profile_qs` instead of `select_related` here?

Reply:
Fixed this 👍